build-samples/build_samples.sh

```diff
@@ -15,6 +15,23 @@ if [ $? -eq 0 ]; then
 	export LLVM="-$LLVM_VER"
 fi
 
+if [[ $(uname -m) != "$TARGETARCH" ]]; then
```

Comment:
I don't think TARGETARCH has been assigned yet. Probably missing TARGETARCH=$4 earlier in the script.